src/topp/MassTraceExtractor.cpp

```diff
@@ -252,6 +252,23 @@ class TOPPMassTraceExtractor :
         fcons.setRT(m_traces_final[i].getCentroidRT());
         fcons.setMZ(m_traces_final[i].getCentroidMZ());
         fcons.setIntensity(m_traces_final[i].getIntensity(false));
+
+        // attach mz and ion mobility as meta data
+        if (m_traces_final[i].fwhm_mz_avg > 0)
+        {
```

Comment:
There is a lot of implicit stuff happening here and at other places with comparisons to some magic number (like -1 or 0).
 As someone reading the code I would wonder why you don't have one variable bool hasIM_, set them once and then use it everywhere where you actually test whether we processed IM data.
This could make reading the code much easier